enhancements/cluster-logging/api-audit-log-policy.md

```diff
@@ -0,0 +1,234 @@
+- vector reads stdout of the filter from the Exec source.
+- from there we route the events as normal
+
+Note: all of the above should support multiple distinct policies, so unique names for policy files etc.
+
+### Risks and Mitigations
+
+#### Data security
+
+No new sensitive data is exposed.
+The `ClusterLogForwarder` can already forward API server audit logs on OSD, ROSA classic and on-premise clusters.
+
+#### Run-time resource costs
+
+OSD and ROSA classic have been doing this type of filtering on managed clusters for years, no new problems are expected.
+
+#### Compliance and legal risks
+
+Incorrectly filtering out important audit events may be a compliance/legal problem.
+The customer creates their own filtering configuration, it needs to be clear that it is their responsibility to do it correctly.
+
+#### Roll-out
+
+Existing customers using the CLO add-on will be unaffected.
+The customer `cl/instance` and `clf/instance` resources will not be modified.
+The new features will be activated only when the customer modifies their configuration as described above.
+
+The existing SRE Splunk exporter SHOULD be replaced by separate CL/CLF resources in a separate namespace as soon as possible.
+Depends on [LOG-1343 Multiple forwarders](https://issues.redhat.com/browse/LOG-1343).
+
+However, the Splunk exporter MAY continue to run alongside customer CLO deployments using these new features, if necessary.
+Both can scrape from the same audit log files simultaneously, and perform separate filtering.
+
+**NOTE**: this is not true for Hypershift, [see hypershift proposal](hypershift-audit-logs.md)
+
+### Drawbacks
+
+- Increased scope for user error.
+- Additional CPU cost, latency (to be measured)
+- New dependency to be maintained for the log collector.
+
+## Design Details
+
+### Open Questions
+
+#### Form of audit policy
+
+The [Kubernetes Audit Policy File](https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/) is expressed as a k8s "object" but not as a typical k8s "resource (no spec/status).
+
+Should we
+- Use it as a resource unmodified
+- Wrap it in a more typical spec/status resource
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+None.
+
+### Upgrade / Downgrade Strategy
+
+Nothing new.
+
+### Version Skew Strategy
+
+The same CLO operator will manage the customer and SRE log forwarding.
+The configuration is entirely separate, but if customer or SRE require new features, the same operator has to be upgraded.
+SRE logging requirements to date are quite stable, and the logging team can ensure they are met by new versions.
+The problem only arises if SRE want to upgrade logging for some new feature or fix, but customer refuses.
+
+### Operational Aspects of API Extensions
+
+SRE and customer CLF instances are separate and independent.
```

Comment:
The CLO installation is shared.  SRE doesn't install CLO right now so there are 2 scenarios:
1. Customer has CLO installed already.
2. Customer does not have CLO already installed.

For the first case we have an issue.  Quite simply, CLO is a customer workload and we will not and cannot manage it.  This is a major operational hurdle and was discussed at length in the context of CMA/KEDA for auto-scaling ingress.  It was decided that we could not support an SRE deployment of CMA.  This is a material risk for the adoption of this solution within SD and should be called out that acceptance of this enhacnement does not guarentee consumption by SD.

Reply:
The customers CLO is deployed on the hosted cluster. SRE's instance is deployed on the management cluster. They are fully separated. The customer has no _direct_ way of influencing the SRE's CLO.
The customer can create a HyperShiftLogForwarder. The HLO on the management cluster combines this with the service providers ClusterLogForwarderTemplate to produce a ClusterLogForwarder on the HCP. This contains config pulled from the hosted HLF, but otherwise it is entirely controlled by the CLO on the management cluster.
So SD has full control over everything running on the management cluster. The management CLO will run collectors on the management cluster, but cost-wise that's no different from running the existing splunk exporter.

Reply:
Context of this comment is not hypershift-audit-logs.md, in classic OSD/ROSA there is no management cluster involved.  There is one CLO on the customer's cluster.